template/golang-http/modules-cleanup.sh

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env sh
+
+set -e
+
+GO111MODULE=$(go env GO111MODULE)
+
+# move_vendor will copy the function's vendor folder,
+# if it exists.
+move_vendor() {
+    if [ ! -d ./function/vendor ]; then
+        echo "vendor not found"
+        return
+    fi
+
+    echo "moving function vendor"
+    mv -f ./function/vendor .
+}
+
+
+# cleanup_gomod will move the function's go module
+cleanup_gomod() {
+
+    # Nothing to do when modules is explicitly off
+    # the z prefix protects against any SH wonkiness
+    # see https://stackoverflow.com/a/18264223
+    if [ "z$GO111MODULE" = "zoff" ]; then
+        echo "modules disabled, skipping go.mod cleanup"
+        return;
+    fi
+
+    if [ ! -f ./function/go.mod ]; then
+        echo "module not initialized, skipping go.mod cleanup"
+        return;
+    fi
+
+    echo "cleaning up go.mod"
+
+    # Copy the user's go.mod
+    mv -f ./function/go.mod .
+    mv -f ./function/go.sum .
```

Comment:
It seems like this fails in case the file is not present, making the docker image construction fail

Reply:
yes it currently assumes that you have a go.sum if you have a go.mod. Per the recommendations from the Go team, the file should exist and should be committed with your project https://github.com/golang/go/wiki/Modules#should-i-commit-my-gosum-file-as-well-as-my-gomod-file

Reply:
Ah ok, that I did not know. Thank you very much, I have modified my functions!